pgcli/packages/pgspecial.py

```diff
@@ -25,6 +25,26 @@ def parse_special_command(sql):
     verbose = '+' in command
     return (command.strip(), verbose, arg.strip())
 
+def list_schemas(cur, pattern, verbose):
+    """
+    Returns (rows, headers, status)
+    """
+    sql = """SELECT n.nspname AS "Name",
+    pg_catalog.pg_get_userbyid(n.nspowner) AS "Owner"
+    FROM pg_catalog.pg_namespace n WHERE n.nspname """
+
+    if pattern:
+        sql += "~ '^(" + pattern.replace('*', '.*') + ")$'"
```

Comment:
sql_name_pattern is not really doing what we need.

First of all \dn accepts only schema name, so we don't need to extract schema-table pair from it.
Then, any object may contain "." in it's name. For example: "my.schema"."my.table". Moreover, object name could also contain even " (double quote) character.
I think we should reimplement sql_name_pattern by using the same approach as in original processSQLNamePattern function: http://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/bin/pg_dump/dumputils.c;h=55509866ff5c692288e8b31074a6912020b381ca;hb=master#l956